Model/Order/PlaceOrder/ProcessOrderPayment.php

```diff
@@ -42,14 +42,14 @@ public function __construct(
      *
      * @param OrderInterface $order
      * @param OrderPaymentInterface $payment
-     * @param TransactionInterface $transaction
+     * @param ?TransactionInterface $transaction
      * @return void
      * @throws Exception
      */
     public function process(
         OrderInterface $order,
         OrderPaymentInterface $payment,
-        TransactionInterface $transaction
+        ?TransactionInterface $transaction
```

Comment:
nit: would we want this parameter to have a default value to null or it's more a specific edge case where it can happen to be null and we want to be able to deal with this?